alerts/resource_alerts.libsonnet

```
@@ -43,9 +43,9 @@
             },
           } else {
             expr: |||
-              sum(namespace_cpu:kube_pod_container_resource_requests:sum{%(ignoringOverprovisionedWorkloadSelector)s}) - (sum(kube_node_status_allocatable{resource="cpu"}) - max(kube_node_status_allocatable{resource="cpu"})) > 0
+              sum(namespace_cpu:kube_pod_container_resource_requests:sum{%(ignoringOverprovisionedWorkloadSelector)s}) - (sum(kube_node_status_allocatable{resource="cpu", %(kubeStateMetricsSelector)s}) - max(kube_node_status_allocatable{resource="cpu"})) > 0
```

```suggestion
              sum(namespace_cpu:kube_pod_container_resource_requests:sum{%(ignoringOverprovisionedWorkloadSelector)s}) - (sum(kube_node_status_allocatable{resource="cpu", %(kubeStateMetricsSelector)s}) - max(kube_node_status_allocatable{resource="cpu", %(kubeStateMetricsSelector)s})) > 0
```

IIUC, shouldn't this be for all instances of KSM's metrics?

---

it makes no difference with `max()` if you have duplicate metrics or not but it does with `sum()`.

---

For the sake of maintaining uniformity throughout the expressions, and to support KSM selectors at all expected instances, I still think we should make this change.

---

thanks @rexagod, make sense- done